malariagen_data/anopheles.py

```diff
@@ -2363,6 +2426,212 @@ def diversity_stats(
+        random_seed: base_params.random_seed = 42,
+    ):
+        # calculate allele counts for each cohort
+        cohort1_counts = self.snp_allele_counts(
+            region=region,
+            sample_sets=None,
+            sample_query=cohort1_query,
+            cohort_size=cohort_size,
+            site_mask=site_mask,
+            site_class=site_class,
+            min_cohort_size=min_cohort_size,
+            max_cohort_size=max_cohort_size,
+            random_seed=random_seed,
+        )
+
+        cohort2_counts = self.snp_allele_counts(
+            region=region,
+            sample_sets=None,
+            sample_query=cohort2_query,
+            cohort_size=cohort_size,
+            site_mask=site_mask,
+            site_class=site_class,
+            min_cohort_size=min_cohort_size,
+            max_cohort_size=max_cohort_size,
+            random_seed=random_seed,
+        )
+
+        # calculate block length for blen
+        n_sites = cohort1_counts.shape[0]  # number of sites
+        block_length = n_sites // n_jack  # number of sites in each block
+
+        # calculate pairwise fst
+        fst_hudson, se_hudson, vb_hudson, _ = allel.blockwise_hudson_fst(
+            cohort1_counts, cohort2_counts, blen=block_length
+        )
+
+        return fst_hudson, se_hudson
+
+    @check_types
+    @doc(
+        summary="""
+            Compute pairwise average Hudson's Fst between a set of specified cohorts.
+        """,
+        returns="""
+            A Dataframe of the pairwise Fst value and standard error (SE) values.
+        """,
+    )
+    def pairwise_average_fst(
+        self,
+        region: base_params.region,
+        cohorts: base_params.cohorts,
+        sample_sets: Optional[base_params.sample_sets] = None,
+        sample_query: Optional[base_params.sample_query] = None,
+        cohort_size: Optional[base_params.cohort_size] = fst_params.cohort_size_default,
+        min_cohort_size: Optional[
+            base_params.min_cohort_size
+        ] = fst_params.min_cohort_size_default,
+        max_cohort_size: Optional[
+            base_params.max_cohort_size
+        ] = fst_params.max_cohort_size_default,
+        n_jack: base_params.n_jack = 200,
+        site_mask: base_params.site_mask = DEFAULT,
+        site_class: Optional[base_params.site_class] = None,
+        random_seed: base_params.random_seed = 42,
+    ):
+        # set up cohort queries
+        cohorts_checked = self._setup_cohorts(
+            cohorts,
+            sample_sets=sample_sets,
+            sample_query=sample_query,
+            cohort_size=cohort_size,
+            min_cohort_size=min_cohort_size,
+        )
+
+        cohort_ids = list(cohorts_checked.keys())
+        cohort_queries = list(cohorts_checked.values())
+        cohort1_ids = []
+        cohort2_ids = []
+        fst_stats = []
+        se_stats = []
+
+        for i in range(len(cohorts_checked)):
+            for j in range(i + 1):
```

Comment:
```suggestion
        n_cohorts = len(cohorts_checked)
        for i in range(n_cohorts):
            for j in range(i + 1, n_cohorts):
```

Bug fix - in the inner loop, need to iterate starting from i+1, up to n_cohorts.

Reply:
...in order to get the computation working on the correct pairs of cohorts, i.e., all pairs run once and avoiding self comparisons.